common/Timer.cc

```diff
@@ -38,6 +42,13 @@ void Timer::cancel() {
     this->canceled = true;
 }
 
+Timer Timer::fork(ConstExprStr name) {
```

Comment:
I added this method to the Timer API so that I can retroactively report a metric with the same start time but a different name.

For example, without `fork`, there's no way to easily report the end-to-end latency of canceled requests.

Reply:
fork seems a very low level name. Clone? Duplicate?

Reply:
`clone` works for me. 👍 I'll make that change after Jake looks at the rest.